logger/utils.go

```diff
@@ -0,0 +1,39 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package logger
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/aws/shim-loggers-for-containerd/debug"
+)
+
+func retryWithExponentialBackoff(operation func() error, initialBackoff time.Duration, maxBackoff time.Duration) error {
+	attempt := 0
+	reachMaxBackoff := false
+	for {
+		err := operation()
+		if err == nil {
+			return nil
+		}
+
+		// Calculate backoff duration with exponential increase and jitter.
+		backoff := float64(initialBackoff) * math.Pow(2, float64(attempt))
+		if backoff > float64(maxBackoff) {
+			backoff = float64(maxBackoff)
+			reachMaxBackoff = true
+		}
+
+		jitter := 0.25 * backoff // Add 25% jitter.
+		sleepDuration := time.Duration(backoff + jitter)
+
+		debug.SendEventsToLog(DaemonName, fmt.Sprintf("Retrying the operation in %v", sleepDuration), debug.ERROR, 0)
+		time.Sleep(sleepDuration)
+		if !reachMaxBackoff {
+			attempt++
```

Comment:
Why do we stop counting the attempt after we reach max backoff. Since we're retrying infinitely shouldn't we keep incrementing attempt?

Reply:
if we keep increasing attempt, the value of backoff below may exceed the int limit
> backoff := float64(initialBackoff) * math.Pow(2, float64(attempt))

We still retry but it will use maxBackoff.

Reply:
Updates - no more retry.